rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -1141,192 +1144,108 @@ protected abstract RMDeployingNode newRMDeployingNode(String name, NodeSource ns
     }
 
     private RMDeployingNode getDeployingNodeWithLock(final String nodeUrl) {
-        return getPersistedInfraVariable(new PersistedInfraVariablesHandler<RMDeployingNode>() {
-            @Override
-            public RMDeployingNode handle() {
-                return deployingNodesMap.get(nodeUrl);
-            }
-        });
+        return getPersistedInfraVariable(() -> this.deployingNodesMap.get(nodeUrl));
     }
 
     protected RMDeployingNode addDeployingNodeWithLockAndPersist(final String nodeUrl,
             final RMDeployingNode deployingNode) {
-        return setPersistedInfraVariable(new PersistedInfraVariablesHandler<RMDeployingNode>() {
-            @Override
-            public RMDeployingNode handle() {
-                RMDeployingNode previousDeployingNode = deployingNodesMap.put(nodeUrl, deployingNode);
-                getPersistedDeployingNodesUrl().add(nodeUrl);
-                return previousDeployingNode;
-            }
+        return setPersistedInfraVariable(() -> {
+            RMDeployingNode previousDeployingNode = this.deployingNodesMap.put(nodeUrl, deployingNode);
+            getPersistedDeployingNodesUrl().add(nodeUrl);
+            return previousDeployingNode;
         });
     }
 
     private RMDeployingNode removeDeployingNodeWithLockAndPersist(final String nodeUrl) {
-        return setPersistedInfraVariable(new PersistedInfraVariablesHandler<RMDeployingNode>() {
-            @Override
-            public RMDeployingNode handle() {
-                RMDeployingNode removedDeployingNode = deployingNodesMap.remove(nodeUrl);
-                getPersistedDeployingNodesUrl().remove(nodeUrl);
-                return removedDeployingNode;
-            }
+        return setPersistedInfraVariable(() -> {
+            RMDeployingNode removedDeployingNode = this.deployingNodesMap.remove(nodeUrl);
+            getPersistedDeployingNodesUrl().remove(nodeUrl);
+            return removedDeployingNode;
         });
     }
 
     private boolean containsDeployingNodeWithLock(final String nodeUrl) {
-        return getPersistedInfraVariable(new PersistedInfraVariablesHandler<Boolean>() {
-            @Override
-            public Boolean handle() {
-                return getPersistedDeployingNodesUrl().contains(nodeUrl);
-            }
-        });
+        return getPersistedInfraVariable(() -> getPersistedDeployingNodesUrl().contains(nodeUrl));
     }
 
     private void clearDeployingNodesWithLockAndPersist() {
-        setPersistedInfraVariable(new PersistedInfraVariablesHandler<Void>() {
-            @Override
-            public Void handle() {
-                deployingNodesMap.clear();
-                getPersistedDeployingNodesUrl().clear();
-                return null;
-            }
+        setPersistedInfraVariable(() -> {
+            this.deployingNodesMap.clear();
+            getPersistedDeployingNodesUrl().clear();
+            return null;
```

Comment:
Remove return null

Reply:
This is unfortunately not possible as it won't compile.